idepix-s2msi/src/main/java/org/esa/snap/idepix/s2msi/operators/cloudshadow/S2IdepixCloudShadowOp.java

```diff
@@ -174,6 +221,34 @@ private int determineSourceResolution(Product product) throws OperatorException
         throw new OperatorException("Invalid product");
     }
 
+    public static double determineResolution(Product product) {
+        int width = product.getSceneRasterWidth();
+        int height = product.getSceneRasterHeight();
+        GeoPos geoPos1 = product.getSceneGeoCoding().getGeoPos(new PixelPos(width / 2, 0), null);
```

Comment:
There exist some classes in SNAP for computing the geographic distance. Like SphericalDistance or EuclideanDistance. They are all derived from DistanceMeasure. You could use it here.

Reply:
Good hint, but I jsut copied that part from another class. It was already implemented like this before. I think it does not have to be part of this PR.